PolicyContext type doesn't include request

## Bug report

### Required System information

- Node.js version: v18.16.0
- NPM version: 9.5.1
- Strapi version: 4.9.1
- Database: MySQL
- Operating system: Mac
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When creating a new custom policy, the type for `PolicyContext` doesn't include keys that should be available like `request`
```typescript
import type { Policy } from "@strapi/strapi/lib/core/registries/policies";

const policy: Policy = (policyContext, config, { strapi }) => {
    //  error TS2339: Property 'request' does not exist on type 'PolicyContext'.
    console.log(policyContext.request);
    
    // ...

    return false;
};

export default policy;
```

### Steps to reproduce the behavior

1. Create a new file at: `./src/api/[apiName]/policies/is-authorized.ts`
2. Copy the above code
3. (optional) Add policy into the routes file at `./src/api/[apiName]/routes/[apiName].ts`

### Expected behavior

There are no Typescript errors and I can access the request object

### Screenshots

N/A

### Code snippets
Example with policyContext keys logged
```typescript
import type { Policy } from "@strapi/strapi/lib/core/registries/policies";

const policy: Policy = (policyContext, config, { strapi }) => {
  // logs this: ["is","type","request","response","app","req","res","originalUrl","state","matched","router","_matchedRoute","captures","params","routerPath","routerName"];
  console.log(Object.keys(policyContext));

  //  error TS2339: Property 'request' does not exist on type 'PolicyContext'.
  console.log(policyContext.request);
  return false;
};

export default policy;
```

### Additional context

[Koa's](https://koajs.com/#context) documentation references request/response. Could be that [PolicyContext](https://github.com/strapi/strapi/blob/294b0f6cfaabd423569ed1d19c6a428a41f45336/packages/core/strapi/lib/core/registries/policies.d.ts#L4) should extend Koa's [ExtendableContext](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/koa/index.d.ts#L720) instead?

Happy to create the PR if that's the case 😄

--- a/packages/core/strapi/lib/core/registries/policies.d.ts
+++ b/packages/core/strapi/lib/core/registries/policies.d.ts
@@ -1,7 +1,7 @@
-import { BaseContext } from 'koa';
+import { ExtendableContext } from 'koa';
 import { Strapi } from '../../';
 
-interface PolicyContext extends BaseContext {
+interface PolicyContext extends ExtendableContext {
   type: string;
   is(name): boolean;
 }

--- a/packages/core/strapi/lib/core/registries/policies.d.ts
+++ b/packages/core/strapi/lib/core/registries/policies.d.ts
@@ -3,7 +3,7 @@ import { Strapi } from '../../';
 
 interface PolicyContext extends ExtendableContext {
   type: string;
-  is(name): boolean;
+  is(name: string): boolean;
 }
 
 export type Policy<T = unknown> = (

